build-system/tasks/release/index.js

```diff
@@ -175,7 +174,13 @@ function discoverDistFlavors_() {
  * @param {string} tempDir full directory path to temporary working directory.
  */
 async function compileDistFlavors_(distFlavors, tempDir) {
-  for (const {flavorType, command} of distFlavors) {
+  for (const {flavorType, command: baseCommand} of distFlavors) {
+    const command =
+      argv.esm === undefined
+        ? `${baseCommand} --esm && ${baseCommand}`
+        : argv.esm
+        ? `${baseCommand} --esm`
+        : baseCommand;
```

Comment:
> I'd imagine the logic can be simplified to:
> 
> ```js
> const command = argv.esm ? `${baseCommand} --esm` : baseCommand;
> ```
> 
> I suppose the release automation will need to be updated before this can be done?

Correct! The eventual goal is to align this with the same behaviour as `amp dist`. However, note that even for `amp dist` you can add `--no-esm` and get the same result as not passing it at all - a feature of the flag parser we're using

Hence the TODO at the end of the file